user_api.go

```diff
@@ -439,24 +468,25 @@ func NewUserAPIWithAccessKey(cfg config.Config, accessKey string) (*UserAPI, err
 		return nil, fmt.Errorf("failed to intialized access key authenticator: %w", err)
 	}
 
-	return initUserAPI(cfg, authenticator)
+	return initUserAPI(cfg, nil, authenticator)
 }
 
 type authenticator interface {
 	Authenticate(r *resty.Request) error
 }
 
-func initUserAPI(cfg config.Config, auth authenticator) (*UserAPI, error) {
+func initUserAPI(cfg config.Config, xPriv *bip32.ExtendedKey, auth authenticator) (*UserAPI, error) {
 	url, err := url.Parse(cfg.Addr)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse addr to url.URL: %w", err)
 	}
 
 	httpClient := restyutil.NewHTTPClient(cfg, auth)
+	txSigner := transactionsigner.New(xPriv)
 	return &UserAPI{
```

Comment:
The nil value for xPriv could potentially lead to a panic. Adding a check or ensuring it's properly initialized would help prevent this issue.